extensions/amp-story/1.0/amp-story.js

```diff
@@ -452,11 +464,45 @@ export class AmpStory extends AMP.BaseElement {
       );
     }
 
+    this.applyDesktopAspectRatioAttribute_();
+
     if (this.maybeLoadStoryDevTools_()) {
       return;
     }
   }
 
+  /**
+   * Grab the desktop-aspect-ratio attribute, clamp the value
+   * between 1/2 and 3/4 aspect ratios, and apply it to the root element.
+   * @private
+   */
+  applyDesktopAspectRatioAttribute_() {
+    if (
+      this.isLandscapeSupported_() ||
+      !this.element.hasAttribute('desktop-aspect-ratio')
+    ) {
+      return;
+    }
+
+    const splittedRatio = this.element
+      .getAttribute('desktop-aspect-ratio')
+      .split(':');
+    if (splittedRatio.length != 2 || splittedRatio[1] == 0) {
```

Comment:
Do we need `splittedRatio[1] == 0`? Would the clamp function ensure it doesn't render beyond the min and max ratios?

Reply:
If it's 0, the line 495 would break since the denominator is 0.